tools-local/Microsoft.DotNet.Build.Tasks.Local/UploadClient.cs

```diff
@@ -142,6 +141,14 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                     var req = new HttpRequestMessage(HttpMethod.Put, blockListUploadUrl);
                     req.Headers.Add(AzureHelper.DateHeaderString, dt1.ToString("R", CultureInfo.InvariantCulture));
                     req.Headers.Add(AzureHelper.VersionHeaderString, AzureHelper.StorageApiVersion);
+                    if(!string.IsNullOrEmpty(DetermineContentTypeBasedOnFileExtension(filePath)))
```

Comment:
Actually, I would structure this like it was previously.  I think that approach makes more sense:

```
private void SetRequestHeadersBasedOnFileType(HttpRequestMessage req, string fileName)
{
  if(Path.GetExtension(filename) == ".svg")
  {
      req.Headers.Add(ContentTypeString, "image/svg+xml");
      req.Headers.Add(CacheControlString, "No-Cache");
  }
  else if (Path.GetExtension(fileName) == ".version")
  {
    req.Headers.Add(ContentTypeString, "text/plain");
  }
}